include/thread_pool/worker.hpp

```diff
@@ -91,11 +92,54 @@ class Worker
 
 namespace detail
 {
+#ifndef KPSR_WITH_FREERTOS
     inline size_t* thread_id()
     {
         static thread_local size_t tss_id = -1u;
         return &tss_id;
     }
+#else
+    enum
+    {
+        thread_id_RD,
+        thread_id_WR
+    };
+
+    inline void thread_id_op(size_t *id, int op)
+    {
+        static std::map<std::thread::id, size_t> thread_id_map;
+        static ReaderWriterLock l;
+
+        if (op == thread_id_RD)
+        {
+            l.lock_shared();
+            *id = thread_id_map.find(std::this_thread::get_id())->second;
+            l.unlock_shared();
+        }
+        else if (op == thread_id_WR)
+        {
+            l.lock();
+            std::thread::id th_id = std::this_thread::get_id();
+            if (thread_id_map.find(th_id) == thread_id_map.end())
+                thread_id_map.insert(std::make_pair(th_id, *id));
+            else
+                thread_id_map[th_id] = *id;
+            l.unlock();
+        }
+    }
+
+    inline size_t thread_id_get()
```

Comment:
What is the advantage of having a two separate functions for get and set and then having them call the same underlying function? We end up adding an extra if/else check which must be executed each time. Can we instead make the map and the lock a static member of the Worker class and then we don't need to have the `op == thread_id_RD/WR` check?